migrations/versions/48ee74b8a7c8_.py

```diff
@@ -0,0 +1,154 @@
+"""Add tokenowner table and move tokenuser data to new table
+
+Revision ID: 48ee74b8a7c8
+Revises: cb6d7b7bae63
+Create Date: 2019-01-09 16:58:03.968193
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '48ee74b8a7c8'
+down_revision = 'cb6d7b7bae63'
+
+
+from alembic import op
+import sqlalchemy as sa
+from flask_sqlalchemy import SQLAlchemy
+from sqlalchemy.schema import Sequence
+from sqlalchemy import orm
+from privacyidea.models import ResolverRealm, TokenRealm
+import sys
+
+
+db = SQLAlchemy()
+
+
+class Realm(db.Model):
+    __tablename__ = 'realm'
+    __table_args__ = {'mysql_row_format': 'DYNAMIC'}
+    id = db.Column(db.Integer, Sequence("realm_seq"), primary_key=True,
+                   nullable=False)
+    name = db.Column(db.Unicode(255), default=u'',
+                     unique=True, nullable=False)
+    default = db.Column(db.Boolean(), default=False)
+    option = db.Column(db.Unicode(40), default=u'')
+
+
+class TokenOwner(db.Model):
+    __tablename__ = 'tokenowner'
+    id = db.Column(db.Integer(), Sequence("tokenowner_seq"), primary_key=True)
+    token_id = db.Column(db.Integer(), db.ForeignKey('token.id'))
+    token = db.relationship('Token', lazy='joined', backref='token_list')
+    resolver = db.Column(db.Unicode(120), default=u'', index=True)
+    user_id = db.Column(db.Unicode(320), default=u'', index=True)
+    realm_id = db.Column(db.Integer(), db.ForeignKey('realm.id'))
+    realm = db.relationship('Realm', lazy='joined', backref='realm_list')
+
+
+class Token(db.Model):
+    __tablename__ = 'token'
+    __table_args__ = {'mysql_row_format': 'DYNAMIC'}
+    id = db.Column(db.Integer, Sequence("token_seq"),
+                   primary_key=True,
+                   nullable=False)
+    serial = db.Column(db.Unicode(40), default=u'',
+                       unique=True,
+                       nullable=False,
+                       index=True)
+    resolver = db.Column(db.Unicode(120), default=u'',
+                        index=True)
+    resolver_type = db.Column(db.Unicode(120), default=u'')
+    user_id = db.Column(db.Unicode(320),
+                       default=u'', index=True)
+
+
+def upgrade():
+    try:
+        op.create_table('tokenowner',
+        sa.Column('id', sa.Integer()),
+        sa.Column('token_id', sa.Integer(), nullable=True),
+        sa.Column('resolver', sa.Unicode(length=120), nullable=True),
+        sa.Column('user_id', sa.Unicode(length=320), nullable=True),
+        sa.Column('realm_id', sa.Integer(), nullable=True),
+        sa.ForeignKeyConstraint(['realm_id'], ['realm.id'], ),
+        sa.ForeignKeyConstraint(['token_id'], ['token.id'], ),
+        sa.PrimaryKeyConstraint('id')
+        )
```

Comment:
Sorry, one last thing: Should we create this table with ``mysql_row_format="DYNAMIC"``? Like for the periodic task tables?
https://github.com/privacyidea/privacyidea/blob/466d562db1167b7e4a8ed2dcf5ff420bf9af5d44/migrations/versions/2c9430cfc66b_.py#L30

Reply:
if the total number of characters in a row exceeds 768 we need this setting.
As of [mysql 5.7.9](https://dev.mysql.com/doc/refman/5.6/en/glossary.html#glos_dynamic_row_format) and [mariadb 10.1.32/10.2.2](https://mariadb.com/kb/en/library/innodb-system-variables/#innodb_default_row_format) the `DYNAMIC` row format is the default.

Reply:
Thanks! Then I guess we don't really need to set it explicitly, because the table is fairly small.